### DISCUSSION TITLE
`v3.0.0-alpha.0`﹣"vite-plugin-monkey must be the `last one` of plugin list": why?

### DISCUSSION THREAD

**Discussion by Malix-Labs** [lisonge/vite-plugin-monkey#57] [Q&A] [answered]
- [Release `v3.0.0-alpha.0`](https://github.com/lisonge/vite-plugin-monkey/releases/tag/v3.0.0-alpha.0):
  > [⋯](https://github.com/lisonge/vite-plugin-monkey/releases/tag/v3.0.0-alpha.0)
  > ## breaking change
  > - vite-plugin-monkey must be the `last one` of plugin list
  > [⋯](https://github.com/lisonge/vite-plugin-monkey/releases/tag/v3.0.0-alpha.0)

  Why?

**Comment by lisonge** [comment, marked as answer]
in fact, vite-plugin-monkey@v2  is also best to be the last item of the list

plugin@v2 use inlineDynamicImport and not support TopLevelAwait
```mermaid
graph LR;
    A(your code)-- vite build library mode -- others plugins -- vite-plugin-monkey  -->B(iife)
```
---

plugin@v3
```mermaid
graph LR;
    A(your code)-- vite build -- others plugins -->B(esm)
    B -- vite-plugin-monkey -- vite build library mode --> C{has TopLevelAwait\nor DynamicImport}
    C -- yes --> D(systemjs)
    C -- no --> E(iife)
```
